tests/math/test_multi_disptach.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Assertion test for multi_dispatch function/decorator
+"""
+from pennylane import numpy as np
+from pennylane import math as fn
+import autoray
+import numpy as onp
+import pytest
+
+    
+tf = pytest.importorskip("tensorflow", minversion="2.1")
+torch = pytest.importorskip("torch")
+
+test_multi_dispatch_stack_data = [
+    [[1.,0.],[2.,3.]],
+    ([1.,0.],[2.,3.]),
+    onp.array([[1.,0.],[2.,3.]]),
+    np.array([[1.,0.],[2.,3.]]),
+    #torch.tensor([[1.,0.],[2.,3.]]),
```

Comment:
Using torch does not work, I am trying to figure out why

Reply:
What sort of error do you get?

Reply:
I obtain `TypeError: stack(): argument 'tensors' must be tuple of Tensors, not Tensor` which seems to be a problem of autoray's numpy.stack itself. The original numpy accepts the torch tensor as input, while the autoray version does not:

```python
import numpy as onp
from autoray import numpy as anp
import tensorflow as tf
import torch

x = torch.tensor([[1,0],[2,3]])
y = tf.constant([[1,0],[2,3]])

print(onp.stack(x), onp.stack(y))
print(anp.stack(x), anp.stack(y))
```
produces the same `TypeError: stack(): argument 'tensors' must be tuple of Tensors, not Tensor` in autoray.

This leads me to ask the more general question: Is this something we can or even want to try to compensate with multi_dispatch or is this a problem of autoray?

Reply:
Thanks @josh146 !

> Is this something we can or even want to try and compensate in our wrapped stack function?
> Probably, yes, since we want to ensure our 'agnostic' stack functions acts the same across all frameworks.

Okay! Just out of curiosity, how would this best be handled? If we added a wrapped and dispatched `stack` somewhere in `/pennylane/math/` we would have to add an exception in case the case that `like = torch` that ensures the input is in the correct format? Or are there more clever ways of doing something like this in general? (I would already be happy with a yes/no answer in case it is more complicated)

> However, this then falls out of the scope of this PR :)

All right, once I get the tests to check out I will mark the PR as ready for review then :)

Reply:
Yep, that is pretty much how we would address this 🙂 Usually there are two things we try:

- If the 'fix' for one of the frameworks is very specific, we simply do

  ```python
  if qml.math.get_interface(x) == "torch":
      ...
  ```

- Sometimes, we can make the fix in a more agnostic manner. In this case, we could probably just do

  ```python
  if not isinstance(x, list):
      x = [x]

  return anp.stack(x)
  ```

  and then the behaviour would be the same for all frameworks.